### PR TITLE
Add more methods to eval_double

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

File: src/eval_double.cpp [hunk 1]
```diff
@@ -85,6 +89,135 @@ class EvalDoubleVisitor : public BaseVisitor<EvalDoubleVisitor> {
...
+        result_ = ::tgamma(tmp);
+    };
+
+    void bvisit(const Constant &x) {
+        if (x.__eq__(*pi)) {
+            result_ = ::atan2(0, -1);
+        } else if (x.__eq__(*E)) {
+            result_ = ::exp(1);
+        } else {
+            throw std::runtime_error("Unknown constant.");
```
Comment (certik) [inline src/eval_double.cpp]: I think it should be possible to evaluate every constant. So I would name the exception " This constant is not implemented yet.", or something like that.

**Comment by certik** [comment]
Besides my comment above, +1 to merge.

The other comment I have is --- why don't we simply create `eval_double2` or `eval_double_single_dispatch` for the single dispatch and always enable it?

Finally, in the tests, why don't we create some kind of a list of expressions with their results, and then just create a simple loop that loops over it, that way you can then just add one line to also test the `eval_double2` function. 

This can be done in a separate PR.

**Comment by isuruf** [comment]
>The other comment I have is --- why don't we simply create eval_double2 or eval_double_single_dispatch for the single dispatch and always enable it?

Yes, we should do that.

>Finally, in the tests, why don't we create some kind of a list of expressions with their results, and then just create a simple loop that loops over it, that way you can then just add one line to also test the eval_double2 function.

That's a great idea. We'll implement it in another PR

**Comment by isuruf** [comment]
Updated the error message.

**Comment by certik** [comment]
Nice. +1 to merge

**Pull request merged**